content/enterprise_influxdb/v1.9/administration/config-data-nodes.md

```diff
@@ -1251,3 +1252,23 @@ Environment variable: `INFLUXDB_TLS_MAX_VERSION`
     > If AE is left enabled while lowering your `max-fetch`, you will initially see
     > higher CPU load as new shard digest files are created.
     > The added load will drop off after shard digests are completed for existing shards. -->
+
+## Query controls
+
+### `[flux-controller]`
+
+#### `query-concurrency = 0`
+Number of queries allowed to execute concurrently.
+
+#### `query-initial-memory-bytes = 0`
+Initial bytes of memory allocated for a query.
+
+#### `query-max-memory-bytes = 0`
+Maximum total bytes of memory allowed for queries.
+
+#### `total-max-memory-bytes = 0`
+Maximum total bytes of memory allowed for queries.
+
+#### `query-queue-size = 0`
+Maximum number of queries allowed in execution queue.
+When queue limit is reached, new queries are rejected.
```

Comment:
```suggestion
#### query-concurrency
Number of queries allowed to execute concurrently.
Default is `0`.

#### query-initial-memory-bytes
Initial bytes of memory allocated for a query.
Default is `0`.

#### query-max-memory-bytes
Maximum total bytes of memory allowed for queries.
Default is `0`.

#### total-max-memory-bytes
Maximum total bytes of memory allowed for queries.
Default is `0`.

#### query-queue-size
Maximum number of queries allowed in execution queue.
When queue limit is reached, new queries are rejected.
Default is `0`.
```

I know you're just following the convention in from the rest of the config file, but at some point we need to go back and update all these headings. They make the heading anchors unnecessarily complication and are prone to change overtime (when defaults are updated). I think we just update these for now and add the task of reformatting all these configuration headings to the backlog.

Other questions:
- Are these the actual defaults? What kind of behavior does the `0` setting result in for each of these?

Reply:
@lesam Can you comment on these values? Are these actual defaults?

> Are these the actual defaults? What kind of behavior does the 0 setting result in for each of these?

Reply:
https://github.com/influxdata/influxdb/pull/21108/files#r628465515 - see validation logic here:

query-queue-size and query-concurrency have to either both be zero or neither be zero (it doesn't make sense to have one limited and not the other).